salesforce/client.bal

```diff
@@ -141,64 +268,206 @@ public isolated client class Client {
     # + extId - External ID value 
     # + fields - Fields to retrieve 
     # + return - JSON result if successful or else `sfdc:Error`
+    # 
+    # # Deprecated
+    # This function is deprecated as it returns a json response.
+    # Use the new and improved `getByExternalId(string sObject, string extIdField, string extId)` function instead.
     @display {label: "Get Record by External ID"}
-    isolated remote function getRecordByExtId(@display {label: "SObject Name"} string sobject, @display 
-                                              {label: "External ID Field Name"} string extIdField, 
+    @deprecated
+    isolated remote function getRecordByExtId(@display {label: "SObject Name"} string sobject, 
+                                              @display {label: "External ID Field Name"} string extIdField, 
                                               @display {label: "External ID"} string extId, 
                                               @display {label: "Fields to Retrieve"} string... fields) 
                                               returns @display {label: "Result"} json|Error {
-        string path = prepareUrl([API_BASE_PATH, SOBJECTS, sobject, extIdField, extId]);
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sobject, extIdField, extId]);
         if fields.length() > 0 {
-            path = path.concat(self.appendQueryParams(fields));
+            path = path.concat(utils:appendQueryParams(fields));
         }
-        json response = check self->getRecord(path);
+        json response = check self.get(path);
         return response;
     }
 
+    # Creates records based on relevant object type sent with json record.
+    #
+    # + sObjectName - SObject name value
+    # + recordPayload - JSON record to be inserted
+    # + return - Created entity ID if successful or else `sfdc:Error`
+    # 
+    # # Deprecated
+    # This function is deprecated as it expects a json payload.
+    # Use the new and improved `create(string sObjectName, record{} recordPayload)` function instead.
+    @display {label: "Create Record"}
+    @deprecated
+    isolated remote function createRecord(@display {label: "SObject Name"} string sObjectName, 
+                                          @display {label: "Record Payload"} json recordPayload) 
+                                          returns @display {label: "Created Entity ID"} string|Error {
+        http:Request req = new;
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName]);
+        req.setJsonPayload(recordPayload);
+        json|http:ClientError response = self.salesforceClient->post(path, req);
+        if response is json {
+            json|error resultId = response.id;
+            if resultId is json {
+                return resultId.toString();
```

Comment:
using  ` toJsonString()` is not user friendly because it provides the response as a JSON representation.  For eg: it will provide a string with double quotes "0015g00000hECccAAG"